### PR TITLE
stream seek error and WS-Trust issue fixes

### PR DESCRIPTION
**Pull request by kpanwar**
stream seek error and WS-Trust issue fixes

### CODE REVIEW

File: src/ADAL.Common/MexParser.cs [hunk 1]
```diff
@@ -141,7 +141,7 @@ internal static WsTrustAddress ExtractWsTrustAddressFromMex(XDocument mexDocumen
                     XElement auth = element.Elements(XmlNamespace.Http + "NegotiateAuthentication").FirstOrDefault();
                     if (auth != null)
                     {
-                        AddPolicy(policies, policy, UserAuthType.IntegratedAuth, securityPolicy.Equals(XmlNamespace.Sp2005));
+                        AddPolicy(policies, policy, UserAuthType.IntegratedAuth, policy.Descendants(XmlNamespace.Sp2005 + "TransportBinding").Any());
```
Comment (weijjia) [inline src/ADAL.Common/MexParser.cs]: Just out of curiosity, what does the AddPolicy do?

**Comment by weijjia** [comment]
:shipit:

File: tests/Test.ADAL.NET.Unit/NonInteractiveTests.cs [hunk 2]
```diff
@@ -128,6 +129,28 @@ public class NonInteractiveTests : AdalTestsBase
...
+            XDocument mexDocument = null;
+            using (Stream stream = new FileStream("TestMex2005.xml", FileMode.Open))
+            {
+                mexDocument = XDocument.Load(stream);
+            }
+
+            Verify.IsNotNull(mexDocument);
+            Dictionary<string, MexPolicy> policies = MexParser.ReadPolicies(mexDocument);
+            Verify.IsNotNull(policies);
+            Verify.IsTrue(policies.Count == 2);
```
Comment (cchar) [inline tests/Test.ADAL.NET.Unit/NonInteractiveTests.cs]: Are you verifying specifically (and separately) for WIA policies and usernamepassword policies? I don't see that, and it seems like you should test for each specifically.

**Comment by cchar** [comment]
:shipit:

**Pull request merged**